src/webR/robj.ts

```diff
@@ -24,60 +24,67 @@ export type RRaw = RProxy<RObjRaw>;
 // RFunction proxies are callable
 export type RFunction = RProxy<RObjFunction> & ((...args: unknown[]) => Promise<unknown>);
 
-export enum RType {
-  Null = 0,
-  Symbol,
-  Pairlist,
-  Closure,
-  Environment,
-  Promise,
-  Call,
-  Special,
-  Builtin,
-  String,
-  Logical,
-  Integer = 13,
-  Double,
-  Complex,
-  Character,
-  Dots,
-  Any,
-  List,
-  Expression,
-  Bytecode,
-  Pointer,
-  Weakref,
-  Raw,
-  S4,
-  New = 30,
-  Free = 31,
-  Function = 99,
-}
-
 export type RPtr = number;
 
-export enum RTargetType {
-  RAW = 'RAW',
-  PTR = 'PTR',
-  ERR = 'ERR',
-}
+export const RType = {
+  null: 0,
+  symbol: 1,
+  pairlist: 2,
+  closure: 3,
+  environment: 4,
+  promise: 5,
+  call: 6,
+  special: 7,
+  builtin: 8,
+  string: 9,
+  logical: 10,
+  integer: 13,
+  double: 14,
+  complex: 15,
+  character: 16,
+  dots: 17,
+  any: 18,
+  list: 19,
+  expression: 20,
+  bytecode: 21,
+  pointer: 22,
+  weakref: 23,
+  raw: 24,
+  s4: 25,
+  new: 30,
+  free: 31,
+  function: 99,
+} as const;
+type RTypeKey = keyof typeof RType;
+type RTypeNumber = typeof RType[keyof typeof RType];
+
+export const RTargetType = {
+  raw: 0,
+  ptr: 1,
+  err: 2,
+} as const;
 
-export type RTargetPtr = {
-  obj: RPtr;
-  methods: string[];
-  type: RTargetType.PTR;
-};
 export type RTargetRaw = {
   obj: RawType;
-  type: RTargetType.RAW;
+  targetType: typeof RTargetType.raw;
```

Comment:
Then if I'm not mistaken this could be:

```suggestion
  targetType: 'raw';
```